src/dbnode/storage/index/convert/convert.go

```diff
@@ -108,22 +119,15 @@ func ValidateSeriesTag(tag ident.Tag) error {
 
 // FromSeriesIDAndTags converts the provided series id+tags into a document.
 func FromSeriesIDAndTags(id ident.ID, tags ident.Tags) (doc.Metadata, error) {
-	clonedID := clone(id)
+	clonedID := clone(id.Bytes())
 	fields := make([]doc.Field, 0, len(tags.Values()))
+	expectedIdx := seriesIDFirstTagBytesIdx
 	for _, tag := range tags.Values() {
 		nameBytes, valueBytes := tag.Name.Bytes(), tag.Value.Bytes()
 
 		var clonedName, clonedValue []byte
-		if idx := bytes.Index(clonedID, nameBytes); idx != -1 {
-			clonedName = clonedID[idx : idx+len(nameBytes)]
-		} else {
-			clonedName = append([]byte(nil), nameBytes...)
-		}
-		if idx := bytes.Index(clonedID, valueBytes); idx != -1 {
-			clonedValue = clonedID[idx : idx+len(valueBytes)]
-		} else {
-			clonedValue = append([]byte(nil), valueBytes...)
-		}
+		clonedName, expectedIdx = findSliceOrClone(clonedID, nameBytes, expectedIdx)
+		clonedValue, expectedIdx = findSliceOrClone(clonedID, valueBytes, expectedIdx)
```

Comment:
Try benchmarking with `nameBytes, valueBytes` inlined (also the same for `FromSeriesIDAndTagIter`). Not much hope here, just want to be sure.

Reply:
Did not see a performance improvement